conn.go

```diff
@@ -229,14 +229,25 @@ func (c *Conn) handleGreet(enhanced bool, arg string) {
 	// NewSession can access it via Conn.Hostname.
 	c.helo = domain
 
-	sess, err := c.server.Backend.NewSession(c)
-	if err != nil {
-		c.helo = ""
-		c.writeError(451, EnhancedCode{4, 0, 0}, err)
-		return
-	}
+	// RFC 5321: "An EHLO command MAY be issued by a client later in the session"
+	if c.session != nil {
+		// RFC 5321: "If it is issued after the session begins
+		// and the EHLO command is acceptable to the SMTP server, ..."
+		if !c.fromReceived {
```

Comment:
Hm, I don't quite get why this condition is here. Why do we need to skip `reset()` when the client has sent `MAIL FROM`?

Reply:
It's what I found in the session that may be usable for the condition "if it is issued after the session begins"

Reply:
["Session begins"](https://datatracker.ietf.org/doc/html/rfc5321#section-3.1) just means that a TCP connection is opened no?

Reply:
Intersting question. If so, then do you think that an EHLO after RSET should not be passed to the session as Reset()?

Reply:
My point is that we should probably always call `c.reset()`, even in the case where `MAIL FROM` was sent by the client (that's what `fromReceived` means).

Reply:
Okay, I can do that. I'll update the PR.

Reply:
But feel free to adapt it to your likings and code style etc.